Receiving error about classpath

When trying to run 
`repomate -p junit4 clone -mn WEEK -rtd TEST-DIRECTORY -v`
I receive 
`[ERROR] _ActException: HookResult(hook='junit4', status=<Status.ERROR: 'error'>, msg='junit-4.12.jar not on the classpath')`
Even though the classpath is set correctly

--- a/repomate_junit4/_junit4_runner.py
+++ b/repomate_junit4/_junit4_runner.py
@@ -57,7 +57,7 @@ def _generate_default_security_policy(classpath: str) -> str:
     junit_jar_matches = re.search(pattern, classpath)
     if not junit_jar_matches:
         raise ValueError("{} not on the classpath".format(JUNIT_JAR))
-    path = junit_jar_matches[0]
+    path = junit_jar_matches.group(0)
     return _DEFAULT_SECURITY_POLICY_TEMPLATE.format(junit4_jar_path=path)
 
 

--- a/repomate_junit4/_junit4_runner.py
+++ b/repomate_junit4/_junit4_runner.py
@@ -52,13 +52,12 @@ def _generate_default_security_policy(classpath: str) -> str:
     """Generate the default security policy from the classpath. ``junit-4.12.jar``
     must be on the classpath.
     """
-    pattern = "{sep}([^{sep}]*{junit_jar}){sep}".format(
-        sep=os.pathsep, junit_jar=JUNIT_JAR
-    )
+    escaped_junit_jar = JUNIT_JAR.replace(".", r"\.")
+    pattern = "[^{sep}]*{junit_jar}".format(sep=os.pathsep, junit_jar=escaped_junit_jar)
     junit_jar_matches = re.search(pattern, classpath)
     if not junit_jar_matches:
         raise ValueError("{} not on the classpath".format(JUNIT_JAR))
-    path = junit_jar_matches.group(1)
+    path = junit_jar_matches[0]
     return _DEFAULT_SECURITY_POLICY_TEMPLATE.format(junit4_jar_path=path)
 
 

--- a/tests/test_junit4.py
+++ b/tests/test_junit4.py
@@ -352,6 +352,26 @@ Expected: is <false>
 
         assert result.status == Status.SUCCESS
 
+    _CP = "{}:{}:{}:{}"
+
+    @pytest.mark.parametrize(
+        "classpath",
+        [
+            _CP.format(JUNIT_PATH, HAMCREST_PATH, "garbage/path", "."),
+            _CP.format(HAMCREST_PATH, "garbage/path", JUNIT_PATH, "."),
+            _CP.format("garbage/path", HAMCREST_PATH, ".", JUNIT_PATH),
+        ],
+    )
+    def test_jars_found_on_classpath(self, setup_hooks, classpath):
+        """Test that acting on a repo when the hamcrest and junit jars are only
+        specified on the classpath works as intended.
+        """
+        hooks = setup_hooks(hamcrest_path="", junit_path="", classpath=classpath)
+
+        result = hooks.act_on_cloned_repo(SUCCESS_REPO)
+
+        assert result.status == Status.SUCCESS
+
 
 class TestSecurityPolicy:
     """Tests that assert that the default security policy model blocks access

